aten/src/ATen/test/extension_backend_test.cpp

```diff
@@ -53,6 +71,25 @@ TEST(BackendExtensionTest, TestRegisterOp) {
   add(a, b);
   ASSERT_EQ(test_int, 3);
 
+  // copy has a double dispatch that needs to be tested
+  register_extension_backend_op(
+    Backend::MSNPU,
+    "s_copy_(Tensor self, Tensor src, bool non_blocking) -> Tensor", &s_copy__override);
+  register_extension_backend_op(
+    Backend::MSNPU,
+    "_s_copy_from(Tensor self, Tensor dst, bool non_blocking) -> Tensor", &_s_copy_from_override);
+  // expand is needed for copy_
+  register_extension_backend_op(
+    Backend::MSNPU,
+    "expand(Tensor self, IntList size, bool implicit) -> Tensor",
+    &expand_override);
+  Tensor cpu_tensor = empty({}, at::kCPU);
+  Tensor msnpu_tensor = empty({}, at::kMSNPU);
+  msnpu_tensor.copy_(cpu_tensor, false);
+  ASSERT_EQ(test_int, 4);
+  cpu_tensor.copy_(msnpu_tensor, false);
+  ASSERT_EQ(test_int, 5);
```

Comment:
what if you copy from msnpu to msnpu?  Do you have to override both of these?  Just _copy_same_type_?

Reply:
also what if you try copying not from CPU and not from same type?  I thought we weren't going to support that.  What happens if you try it?  These should all be tested IMO so there is logical documentation people can look at.